arch/arm/soc/arm/beetle/soc.h

```diff
@@ -89,6 +89,10 @@
 
 #ifndef _ASMLANGUAGE
 
+/* ARM CMSIS definitions must be included before kernel_includes.h.
+ * Therefore, it is essential to include kernel_includes.h after including
+ * core SOC-specific headers.
+ */
```

Comment:
Is it possible to enforce this somehow?

e.g.

in kernel_includes.h

do

```
#ifndef ARM_CMSIS_DEFINITIONS
#error "ARM CMSIS definitions must be included before kernel_includes.h"
#endif
```

and vice-versa for soc.h.

Reply:
Hm, I would not want to pollute kernel_includes.h, as this is a cross-architecture header.

Reply:
I agree that kernel_includes.h should not reference specific arch's.

e.g. adding a new arch should not involve a patch to kernel_includes.h.

But using a cross-platform reference like "soc" is OK. If it is truly correct that kernel_includes.h depends on soc specific symbols.

Reply:
how about adding the above check in include/arch/arm/arch.h?